src/bijectors/distribution_bijector.jl

```diff
@@ -1,4 +1,3 @@
-# TODO: Deprecate?
```

Comment:
At this point, `DistributionBijector` should really be deprecated. The idea was to use it as a placeholder while we implemented `Bijector` for all the different distributions. 

In this PR the goal is to basically make `link` and `invlink` nothing but wrappers around the `Bijector` interface. This then makes `DistributionBijector` completely unnecessary:)

Reply:
Well for `(Fill)(Vector|Matrix)Of(Univariate|Multivariate)` distributions, somewhat efficient `logpdf_with_trans`, `link` and `invlink` methods are there which are better than falling back on `Stacked`. `Stacked` wasn't really designed for this, think 1000s of distributions of the same type and all are univariate. The whole `ranges` field is unnecessary, not to mention it is a tuple, so the length is part of the type. We can make a different bijector for these distributions.

Reply:
Can't the `ranges` field just be a single-element then, i.e. `1:end`?
And combine this with a specific impl when `dists` are `Fill`, we should obtain the same perf, right?

Reply:
Hmm I may have misinterpreted the `ranges` field! Let me take a look again in a bit.